EasyGYM/Data/WorkoutService.swift

```diff
@@ -26,10 +28,17 @@ class WorkoutService {
                 let name = data["name"] as? String ?? ""
                 
                 let muscle = data["muscle"] as? [String: Any]
-                let primaryMuscle = muscle?["Primary"] as? [String] ?? []
-                let secondaryMuscle = muscle?["Secondary"] as? [String] ?? []
+                let primaryMuscle = (muscle?["Primary"] as? [String])?.compactMap { MuscleType(rawValue: $0.lowercased()) } ?? []
+                let secondaryMuscle = (muscle?["Secondary"] as? [String])?.compactMap { MuscleType(rawValue: $0.lowercased()) } ?? []
```

Comment:
there are a lot of fallbacks. 
With the change of Workout conforming to Codable it will not be needed to create Workouts this way

use generic methods to get to your wanted outputs. these can live in the class that holds all your methods for fetching stuff from firebase.
f.e. 
```
func fetchDocument<T: Codable>(collectionName: String, documentID: String, completion: @escaping (Result<T, Error>) -> Void) {
    let db = Firestore.firestore()
    
    db.collection(collectionName).document(documentID).getDocument { (document, error) in
        if let error = error {
            completion(.failure(error))
            return
        }
        
        guard let document = document, document.exists else {
            completion(.failure(NSError(domain: "", code: -1, userInfo: [NSLocalizedDescriptionKey: "Document does not exist"])))
            return
        }
        
        do {
            if let data = document.data() {
                let decodedObject = try Firestore.Decoder().decode(T.self, from: data)
                completion(.success(decodedObject))
            } else {
                completion(.failure(NSError(domain: "", code: -1, userInfo: [NSLocalizedDescriptionKey: "No data found"])))
            }
        } catch let error {
            completion(.failure(error))
        }
    }
}
```

or when its an Array

```
func fetchDocuments<T: Codable>(collectionName: String, completion: @escaping (Result<[T], Error>) -> Void) {
    let db = Firestore.firestore()
    
    db.collection(collectionName).getDocuments { (querySnapshot, error) in
        if let error = error {
            completion(.failure(error))
            return
        }
        
        guard let documents = querySnapshot?.documents else {
            completion(.failure(NSError(domain: "", code: -1, userInfo: [NSLocalizedDescriptionKey: "No documents found"])))
            return
        }
        
        do {
            var objects: [T] = []
            for document in documents {
                let data = document.data()
                let decodedObject = try Firestore.Decoder().decode(T.self, from: data)
                objects.append(decodedObject)
            }
            completion(.success(objects))
        } catch let error {
            completion(.failure(error))
        }
    }
}
```